DRAFT:  Simplify  permissions and do not short circuit

kcristiano opened the pull request:
Overview
----------------------------------------
Clean up permissions for WP and remove all short circuits

Before
----------------------------------------

There were a number of places where permissions were bypassed if criteria was met.

After
----------------------------------------
No bypass

Technical Details
----------------------------------------
This should not be merged until extensive testing is done.   Particular attention must be paid to WP Multi-Site.

kcristiano committed:
Draft at cleaning up permissions

kcristiano committed:
Further permission tweaks

kcristiano commented:
added a second commit to bypass permissions for `Administrator` role and `SuperAdmin`.

Original test fails were valid and administrator did not have enough permissions.

rbaugh commented:
@kcristiano The reason for the anonymous check failures is the removal of the `$userId !== 0`. In most of the other calls to the WP check function, the `$userId` is null but in the [anonymous check](https://github.com/civicrm/civicrm-core/blob/master/CRM/Utils/Check/Component/Security.php#L337) it is actually passing in `0` for the `$userId`.

By removing the `$userId` check, it then fallsback to the `wp_current_user` check which ideally could lead to a true response if the person doing the status check had `super admin` or `administrator` capability.

Overall, the permission check would be accurate as it would default to the anonymous user from the `wp_current_user()` call and the permissions wouldn't allow for the Contribution access, but it is just that the status check in CiviCRM would normally only be run by a logged in user who could have the access based on their account.

kcristiano committed:
add back code we actually need

TODO: lines 35-43 may be able to be removed as this was put in place for a 4.5 Paypal issue

kcristiano commented:
Thanks @rbaugh good catch.   Test failure will need to be address, and we will need to do a great deal of testing.

In order to allow the Administrators to have access we are  bypassing permissions for Administrator role and Super Admins.

We could look at explicitly assigning permisions/capabilities, but @colemanw mentione that this would not handle virtual permissions.

I am reluctant to remove the > 10yr old fix for PayPal (CRM-15629) but I don't believe we need this any longer. 

Any suggestions or thoughts?

cc @christianwach
